clients/csharp-uwp/docs/sample-app-walkthrough.md

```diff
@@ -0,0 +1,106 @@
+
+The following steps describe how to complete keyword verification with DLS through the Speech SDK and how to interact with MVA during verification.
+
+### 8. Retrieve activation audio
+Create an [AudioGraph](https://docs.microsoft.com/en-us/uwp/api/windows.media.audio.audiograph) and pass it to the CreateAudioDeviceInputNodeAsync of the ConversationalAgentSession. This will load the graph's audio buffer with the audio *starting approximately 3 seconds before MVA detected a keyword*. In the sample app, [AgentAudioInputProvider](https://github.com/Azure-Samples/Cognitive-Services-Voice-Assistant/blob/master/clients/csharp-uwp/UWPVoiceAssistantSample/AudioInput/AgentAudioInputProvider.cs)'s InitializeFromAgentSessionAsync method retrieves the audio from the ConversationalAgentSession. AgentAudioInputProvider fires an event, DataAvailable, when the audio is ready as a stream of bytes. 
+
+Note: In the sample app, the first 2 seconds of the audio are [trimmed](https://github.com/Azure-Samples/Cognitive-Services-Voice-Assistant/blob/master/clients/csharp-uwp/UWPVoiceAssistantSample/DialogManager.cs#L308) before being sent to Speech Services. This is because the keyword verification may fail if the full 3 seconds from pre-detection are included in the audio.
+
+### 9. Verify the keyword
+Using Cognitive Speech Services or your own system, process the activation audio to verify whether it contains a keyword. The sample app uses the Speech SDK to interface with Speech Services. To use a different keyword verification system, direct the audio from the AgentAudioInputProvider's DataAvailable event to the alternative service and trigger step 10 on keyword confirmation. 
+
+The following goes into further detail on how the sample application uses Direct Line Speech for keyword verification.
+#### Initiate conversation start on signal from MVA
+The DialogManager class manages the lifecycle of a bot interaction, accepting commands to start keyword verification and speech recognition and emitting events with recognition results and bot responses. In step 7, after receiving a signal from MVA, the sample app calls [DialogManager](https://github.com/Azure-Samples/Cognitive-Services-Voice-Assistant/blob/master/clients/csharp-uwp/UWPVoiceAssistantSample/DialogManager.cs).HandleSignalDetection, which initiates keyword verification. 
```

Comment:
I'd tried to (unofficially and informally) start a standard way of describing each of our stages:

1. Detection
2. Confirmation
3. Verification

What are your thoughts on that? I mention it because, here, "verification" conflates with KWV for me. That might not be an issue to the uninitiated, though.

Reply:
You mean for 1st/2nd/3rd stage detection? It may be confusing to say "When the keyword is detected, verify that the keyword is valid... then confirm it is valid". Seems like keeping verification but adding the source (signal detection, local verification, cloud verification) would be less confusing to me.